receiver/redisreceiver/metric_functions.go

```diff
@@ -44,22 +44,49 @@ func (rs *redisScraper) dataPointRecorders() map[string]interface{} {
 		"uptime_in_seconds":               rs.mb.RecordRedisUptimeDataPoint,
 		"used_cpu_sys":                    rs.recordUsedCPUSys,
 		"used_cpu_sys_children":           rs.recordUsedCPUSysChildren,
-		"used_cpu_user":                   rs.recordUsedCPUSysUser,
+		"used_cpu_sys_main_thread":        rs.recordUsedCPUSysMainThread,
+		"used_cpu_user":                   rs.recordUsedCPUUser,
+		"used_cpu_user_children":          rs.recordUsedCPUUserChildren,
+		"used_cpu_user_main_thread":       rs.recordUsedCPUUserMainThread,
 		"used_memory":                     rs.mb.RecordRedisMemoryUsedDataPoint,
 		"used_memory_lua":                 rs.mb.RecordRedisMemoryLuaDataPoint,
 		"used_memory_peak":                rs.mb.RecordRedisMemoryPeakDataPoint,
 		"used_memory_rss":                 rs.mb.RecordRedisMemoryRssDataPoint,
 	}
 }
 
+/**
+ * Redis cpu used time section
+ * Redis versions < 6.0 have:
+ *   used_cpu_sys: System CPU consumed by the Redis server, which is the sum of system CPU consumed by all threads of the server process (main thread and background threads)
+ *   used_cpu_user: User CPU consumed by the Redis server, which is the sum of user CPU consumed by all threads of the server process (main thread and background threads)
+ *   used_cpu_sys_children: System CPU consumed by the background processes
+ *   used_cpu_user_children: User CPU consumed by the background processes
+ * Redis versions >= 6.0 have two more:
+ *   used_cpu_sys_main_thread: System CPU consumed by the Redis server main thread
+ *   used_cpu_user_main_thread: User CPU consumed by the Redis server main thread
+ */
```

Comment:
Can we convert the attribute into enum and document this in metadata.yaml?

Reply:
done